languages/python/sqlalchemy-oso/sqlalchemy_oso/roles.py

```diff
@@ -1,16 +1,142 @@
 from typing import Any, List
 
 from sqlalchemy.types import Integer, String
-from sqlalchemy.schema import Table, Column, ForeignKey
+from sqlalchemy.schema import Column, ForeignKey
 from sqlalchemy.ext.declarative import declared_attr
-from sqlalchemy.orm import relationship, backref
-from sqlalchemy import inspect
-
+from sqlalchemy.orm import relationship, validates
+from sqlalchemy import inspect, UniqueConstraint
+from .session import _OsoSession
 
 ROLE_CLASSES: List[Any] = []
```

Comment:
Can we add a comment saying what this is used for.

From what I can tell, this is just used to construct the policy helpers, right?

Reply:
This is used to generate the rbac base policy. It stores the resource and role model inputs for each role model as it is created, which we use to write the specializers in the base policy.

Just thinking about this more now, we don't check for the case in which a user creates multiple role models for the same resource. We should probably fix that.

Reply:
Oh just saw your request about a comment, will add now